torch/signal/windows/windows.py

```diff
@@ -0,0 +1,303 @@
+
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+window_common_args = merge_dicts(
+    parse_kwargs(
+        """
+    M (int): the length of the window.
+        In other words, the number of points of the returned window.
+    sym (bool, optional): If `False`, returns a periodic window suitable for use in spectral analysis.
+        If `True`, returns a symmetric window suitable for use in filter design. Default: `True`.
+"""
+    ),
+    factory_common_args,
+    {"normalization": "The window is normalized to 1 (maximum value is 1). However, the 1 doesn't appear if "
+                      "`M` is even and `sym` is `True`."}
+)
+
+
+def _add_docstr(*args):
+    r"""Adds docstrings to a given decorated function.
+
+    Specially useful when then docstrings needs string interpolation, e.g., with
+    str.format().
+    REMARK: Do not use this function if the docstring doesn't need string
+    interpolation, just write a conventional docstring.
+
+    Args:
+        args (str):
+    """
+    def decorator(o):
+        o.__doc__ = "".join(args)
+        return o
+
+    return decorator
+
+
+def _window_function_checks(function_name: str, M: int, dtype: torch.dtype, layout: torch.layout) -> None:
+    r"""Performs common checks for all the defined windows.
+     This function should be called before computing any window.
+
+     Args:
+         function_name (str): name of the window function.
+         M (int): length of the window.
+         dtype (:class:`torch.dtype`): the desired data type of returned tensor.
+         layout (:class:`torch.layout`): the desired layout of returned tensor.
+     """
+    if M < 0:
+        raise ValueError(f'{function_name} requires non-negative window length, got M={M}')
+    if layout is not torch.strided:
+        raise ValueError(f'{function_name} is implemented for strided tensors only, got: {layout}')
+    if not is_float_dtype(dtype):
+        raise ValueError(f'{function_name} expects floating point dtypes, got: {dtype}')
+
+
+@_add_docstr(
+    r"""
+Computes a window with an exponential waveform.
+Also known as Poisson window.
+
+The exponential window is defined as follows:
+
+.. math::
+    w(n) = \exp{\left(-\frac{|n - c|}{\tau}\right)}
+
+where `c` is the center of the window.
+    """,
+    r"""
+
+{normalization}
+
+Args:
+    {M}
+
+Keyword args:
+    center (float, optional): where the center of the window will be located.
+        Default: `M / 2` if `sym` is `False`, else `(M - 1) / 2`.
+    tau (float, optional): the decay value.
+        Tau is generally associated with a percentage, that means, that the value should
+        vary within the interval (0, 100]. If tau is 100, it is considered the uniform window.
+        Default: 1.0.
+    {sym}
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples::
+    >>> # Generate a symmetric exponential window of size 10 and with a decay value of 1.0.
```

Comment:
I know what it's missing! Damn sphinx, it's missing a new line!
```suggestion
Examples::

    >>> # Generate a symmetric exponential window of size 10 and with a decay value of 1.0.
```

That should render correctly.

Sphinx is the worst. Even worse, I think we are using a fairly old version. Probably a newer version has already fixed this issue, and you may have a newer version installed locally.

Reply:
Lol, that's a tricky one. Thanks!